src/main/java/Ken/task/TaskList.java

```diff
@@ -0,0 +1,223 @@
+
+    /**
+     * Gets the list of tasks.
+     *
+     * @return The list of tasks.
+     */
+    public List<Task> getTasks() {
+        return this.tasks;
+    }
+
+    /**
+     * Adds a task to the list.
+     *
+     * @param task The task to be added.
+     */
+    public void addTask(Task task) {
+        if (tasks.size() < MAX_TASKS) {
+            tasks.add(task);
+            System.out.println("Got it!");
+            System.out.println("added task: " + task);
+            System.out.println("Now Barbie has " + tasks.size() + " tasks in list\n");
+
+        } else {
+            System.out.println("Way too many tasks for today Barbie!");
+            System.out.println("Slow the Slayy\n");
+
+        }
+    }
+
+    /**
+     * Adds a todo task to the list.
+     *
+     * @param description The description of the todo task.
+     * @throws KenException If the description is empty.
+     */
+    public void addTodoTask(String description) throws KenException {
+        if (description.isEmpty()) {
+            throw new KenException("do what?");
+        }
+        Todo todo = new Todo(description);
+        addTask(todo);
+    }
+
+    /**
+     * Adds a deadline task to the list.
+     *
+     * @param description The description of the deadline task.
+     * @throws KenException If the description or deadline command is invalid.
+     */
+    public void addDeadlineTask(String description) throws KenException {
+        try {
+            int indexOfBy = description.indexOf("/by");
+            if (indexOfBy != -1) {
+                String deadlineDescription = description.substring(0, indexOfBy).trim();
+                String by = description.substring(indexOfBy + 3).trim();
+                Deadline deadline = new Deadline(deadlineDescription, by);
+                addTask(deadline);
+            } else {
+                System.out.println("That's not how you declare a deadline. p.s. use /by.");
+                throw new KenException("Invalid deadline command. By when?.");
+            }
+        } catch (Exception e) {
+            throw new KenException("Invalid deadline command. By when?.");
+        }
+    }
+
+
+    /**
+     * Adds an event task to the list.
+     *
+     * @param description The description of the event task.
+     * @throws KenException If the description or event command is invalid.
+     */
+    public void addEventTask(String description) throws KenException {
+        try {
+            int indexOfFrom = description.indexOf("/from");
+            int indexOfTo = description.indexOf("/to");
+
+            if (indexOfFrom != -1 && indexOfTo != -1) {
+                String eventDescription = description.substring(0, indexOfFrom).trim();
+                String from = description.substring(indexOfFrom + 5, indexOfTo).trim();
+                String to = description.substring(indexOfTo + 3).trim();
+                Event event = new Event(eventDescription, from, to);
+                addTask(event);

```

Comment:
I didn't check the logic, but is it guaranteed that there will be spaces before the `/from` and `/to`? 
Eg do you consider `todos taskName/byNOW/toNOW` a valid input?
This is up to you, just in case you didn't consider it